paddle/fluid/framework/ir/graph.cc

```diff
@@ -175,6 +276,56 @@ void Graph::ResolveHazard(
   }
 }
 
+#if FLAGS_ssa_program
+std::shared_ptr<Graph> Graph::Clone() {
+  PADDLE_ENFORCE_EQ(
+      this->parent_, nullptr,
+      platform::errors::InvalidArgument(
+          "This graph is a subgraph, and can't be cloned individually"));
+  auto cloned_graph = std::make_shared<Graph>(this->program_);
+  cloned_graph->ReleaseSubgraphs();
```

Comment:
Is making a Graph from Program enough? If it is enough, we don't have to release and add here.

Reply:
I think the graph could be changed by some optimization pass, and is different form the initial graph constructed from program. So I release and add subgraph here.